### ISSUE TITLE
Meta::getTables returns tables of type TableType.TABLE when they're actually system views

### ISSUE THREAD

**Issue by MMiljak** [jOOQ/jOOQ#14574]
### Expected behavior

When getting all tables and filtering by table type (now manually, instead of calling isTable()) filter should return only real tables not catalogs and other kinds of objects.

Basically this code should work, it's a bit extreme, but in theory :)

```
dslContext.meta().getTables().forEach(
                t -> {
                    if (t.getTableType().equals(TableType.TABLE)) {
                        dslContext.truncateTable(t).cascade().execute();
                    }
                });
```

Also would be nice to add a filter in TableType for regular tables so that there is a minimal regression introduced (just a different method call)

### Actual behavior

TableType.TABLE includes system tables and catalogs.

```
SQL [truncate table "information_schema"."_pg_foreign_data_wrappers" cascade]; ERROR: "_pg_foreign_data_wrappers" is not a table
org.jooq.exception.DataAccessException: SQL [truncate table "information_schema"."_pg_foreign_data_wrappers" cascade]; ERROR: "_pg_foreign_data_wrappers" is not a table
```

Ignoring system tables starting with '_'
`if (t.getTableType().equals(TableType.TABLE) && !t.getName().startsWith("_")) `

```
SQL [truncate table "information_schema"."administrable_role_authorizations" cascade]; ERROR: "administrable_role_authorizations" is not a table
org.jooq.exception.DataAccessException: SQL [truncate table "information_schema"."administrable_role_authorizations" cascade]; ERROR: "administrable_role_authorizations" is not a table
```

### Steps to reproduce the problem

Execute the following line of code 

```
dslContext.meta().getTables().forEach(
                t -> {
                    if (t.getTableType().equals(TableType.TABLE)) {
                        dslContext.truncateTable(t).cascade().execute();
                    }
                });
```

### jOOQ Version

>= 3.17.6 (didnt test older ones)

### Database product and version

POSTGRESQL 13.8

### Java Version

17.0.2-tem

### OS Version

debian:bullseye

### JDBC driver name and version (include name if unofficial driver)

42.5.2

**Comment by lukaseder** [comment]
Thanks for your report. I can see that pgjdbc returns:

```
+---------+-----------+---------------------------+-------------+-------+--------+----------+---------+-------------------------+--------------+
|TABLE_CAT|TABLE_SCHEM|TABLE_NAME                 |TABLE_TYPE   |REMARKS|TYPE_CAT|TYPE_SCHEM|TYPE_NAME|SELF_REFERENCING_COL_NAME|REF_GENERATION|
+---------+-----------+---------------------------+-------------+-------+--------+----------+---------+-------------------------+--------------+
|{null}   |informat...|_pg_foreign_data_wrappers  |SYSTEM VIEW  |{null} |        |          |         |                         |              |
...
```

So, there's just a disconnect between what we expected (`SYSTEM_VIEW`) and what is being returned (`SYSTEM VIEW`).

I'd like for this issue to fix only this particular thing, which is clearly a defect. You mentioned also a change request to support "regular tables", but I'm not sure if I understand that. Can you please create a new, separate issue for that change request, with some details about what you're expecting there?

In the meantime, I'll fix this `SYSTEM VIEW` bug.

**Comment by lukaseder** [comment]
A workaround is, of course, to filter the schema. You probably don't want to tamper with the system schemas anyway, even if they did contain tables.

**Comment by MMiljak** [comment]
Best workaround is to not uplift, but to stay on 3.17.5 :D

Regarding the "regular table" i was just thinking since TableType has isTable() which now checks for TABLE and TEMPORARY_TABLE, to add a isRegularTable() and isSystemTable() or whatever name you fancy, its just a suggestion, not like i really need it.

Thanks for the quick responses.

**Comment by lukaseder** [comment]
> Best workaround is to not uplift, but to stay on 3.17.5 :D

Ah I see, it is a regression, indeed.

> Regarding the "regular table" i was just thinking since TableType has isTable() which now checks for TABLE and TEMPORARY_TABLE, to add a isRegularTable() and isSystemTable() or whatever name you fancy, its just a suggestion, not like i really need it.

Maybe, in the future. The code generator knows about system tables, maybe, the runtime could, as well.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.8 (https://github.com/jOOQ/jOOQ/issues/14575), and 3.16.14 (https://github.com/jOOQ/jOOQ/issues/14576)

**Comment by MMiljak** [comment]
Wow, damn, fast work.
Thanks a lot.
